src/vs/workbench/contrib/files/browser/views/explorerViewer.ts

```diff
@@ -1362,6 +1363,23 @@ export class FileDragAndDrop implements ITreeDragAndDrop<ExplorerItem> {
 		}
 	}
 
+	private async mergeDirectories(sources: ExplorerItem[], targetResource: URI): Promise<ResourceFileEdit[]> {
+		const resourceFileEdits: ResourceFileEdit[] = [];
+		for (const source of sources) {
+			const newResource = joinPath(targetResource, source.name);
+			// If the directory matches then we should recurse on the children because we want to merge them
+			if (source.isDirectory && await this.fileService.exists(newResource)) {
+				// Recurse into the children of the directories so that we can merge them with the source
+				resourceFileEdits.push(...await this.mergeDirectories(Array.from(source.children.values()), newResource));
```

Comment:
Will `source.children` work even when the folder is not expanded in the Explorer? My understanding would be that we would have to also resolve explorer items recursively to ensure we have all the children recursively?

Reply:
Good catch, I now use the async method `fetchChildren` which should actually resolve them in the model.